cachi2/interface/cli.py

```diff
@@ -346,6 +355,66 @@ def inject_files(
     )
 
 
+def _prevalidate_sbom_files_args(sbom_files_to_merge: Paths) -> Paths:
+    def enough_files_for_merge(sbom_files_to_merge: Paths) -> Paths:
+        if len(sbom_files_to_merge) < 2:
+            raise typer.BadParameter("Need at least two different SBOM files")
+        return sbom_files_to_merge
+
+    def all_files_are_jsons(sbom_files_to_merge: Paths) -> Paths:
+        for sbom_file in sbom_files_to_merge:
+            try:
+                json.loads(sbom_file.read_text())
+            except ValueError:
+                raise typer.BadParameter(f"{sbom_file} does not look like a SBOM file")
+        return sbom_files_to_merge
+
+    return all_files_are_jsons(enough_files_for_merge(list(set(sbom_files_to_merge))))
+
+
+@app.command()
+@handle_errors
+def merge_sboms(
+    sbom_files_to_merge: Paths = typer.Argument(
+        ...,
+        callback=_prevalidate_sbom_files_args,
+        exists=True,
+        file_okay=True,
+        dir_okay=False,
+        resolve_path=True,
+        readable=True,
+        help="Names of files with SBOMs to merge.",
+    ),
+    output_sbom_file_name: Optional[Path] = OUTFILE_OPTION,
+) -> None:
+    """Merge two or more SBOMs into one.
+
+    The command works with Cachi2-generated SBOMs only. You might want to run
+
+    cachi2 fetch-deps <args...>
+
+    first to produce SBOMs to merge.
+    """
+    sboms_to_merge = []
+    for sbom_file in sbom_files_to_merge:
+        try:
+            sboms_to_merge.append(Sbom.model_validate_json(sbom_file.read_text()))
+        except pydantic.ValidationError:
+            raise typer.BadParameter(f"{sbom_file} does not appear to be a valid Cachi2 SBOM.")
```

Comment:
I believe we're avoiding raising other errors besides the ones defined in [errors.py](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/errors.py). The downside is that we lose the nice default Typer formatting in the terminal (although our main real use cases would only see these results in pipeline logs, so the formatting would not matter much).

This is because we wanted to set up a small framework to, besides providing the error message, also provide some potential [solution](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/errors.py#L17) so the user could find his way around it.

Reply:
Fair enough, I'll change exceptions to cachi2 exceptions. Raising them in CLI does produce some long and colorful tracebacks in terminal. I usually tend to hide tracebacks from users, but this is a special case.

Reply:
Changed to proper exceptions in the follow-up revision, resolving.